vassal-app/src/main/java/VASSAL/build/module/map/MoveCameraButton.java

```diff
@@ -124,6 +126,22 @@ private Point getDestination() {
   private void moveCamera() {
     final Point dest = getDestination();
     if (dest != null) {
+      final String zoomString = zoom.getText(map, this, "Editor.MoveCameraButton.zoom");
+      if (!zoomString.strip().isEmpty()) {
+        try {
+          final double toZoom = Double.parseDouble(zoomString);
+          if (toZoom > 0) {
+            final Zoomer zoomer = map.getZoomer();
+            if (zoomer != null) {
+              zoomer.setZoomFactor(toZoom);
+            }
+          }
+        }
+        catch (NumberFormatException e) {
+          ErrorDialog.dataWarning(new BadDataReport(this, Resources.getString("Error.non_number_error"), Resources.getString("Editor.MoveCameraButton.zoom") + ": " + zoomString, e));
+        }
```

Comment:
The try/catch should be tighter around the thing which is intended to throw. What you've got here will catch a `NFE` from anything inside the `try`, but if anything other than `parseDouble` throws one, it's a legitimate bug, not a bad value, and we'd never see it this way.

Reply:
fixed